addons/website/models/res_company.py

```diff
@@ -27,6 +39,14 @@ def google_map_link(self, zoom=8):
         partner = self.sudo().partner_id
         return partner and partner.google_map_link(zoom) or None
 
+    def _get_public_users_count_for_website(self):
+        self.ensure_one()
+        # We need sudo to be able to see public users from others companies too
+        public_users = self.env.ref('base.group_public').sudo().with_context(active_test=False).users
+        public_users_for_website = public_users.filtered(lambda user: user.company_id == self)
+
+        return len(public_users_for_website)
```

Comment:
Probably code to remove (see other comment), but to let you know: instead of copy pasting those 2 lines (containing sudo & comment), I'd extract those 2 lines from `_get_public_user` it in a new method and call it from your new `_check_active` and from the existing `_get_public_user`